src/extensions/replay/sessionrecording-utils.ts

```diff
@@ -165,27 +166,154 @@ export function truncateLargeConsoleLogs(_event: eventWithTime) {
 
 export const SEVEN_MEGABYTES = 1024 * 1024 * 7 * 0.9 // ~7mb (with some wiggle room)
 
-// recursively splits large buffers into smaller ones
+function sliceList(list: any[], sizeLimit: number): any[][] {
```

Comment:
if the list length is 2 and they are both bigger than the limit what does this do

Reply:
Should we possibly try recursively slicing the chunks in https://github.com/PostHog/posthog-js/pull/1307/files#diff-deb180b570d4a9e728cd83fd418e54a748f69492d07c9809924a3821ca4de4d5R178 as we used to do in https://github.com/PostHog/posthog-js/pull/1307/files#diff-deb180b570d4a9e728cd83fd418e54a748f69492d07c9809924a3821ca4de4d5L172-L188

Reply:
so, i used to split in two, and then recursively split each half in two until every part was below the size limit

now we calculate how many chunks so that ever chunk is under the limit and then split into that many chunks

(recursion always feels too "clever" to me :))